### PR TITLE
Pw 4: feat: Add React Material UI library, add responsive navigation menu, and finish home page styling

### PR DESCRIPTION
**Pull request by neonseal**
Things got real wonky with commits. Basically I fucked it up, but I fixed it.... Home page is done!

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -85,7 +85,7 @@ module.exports = {
 		'@typescript-eslint/consistent-type-definitions': 'warn',
 		'@typescript-eslint/no-empty-function': 'warn',
 		'@typescript-eslint/no-empty-interface': 'warn',
-		'@typescript-eslint/no-floating-promises': 'error',
+		'@typescript-eslint/no-floating-promises': 'warn',
```
Comment (WhiterDig) [inline .eslintrc.js]: Was this because of the ` reportWebVitals()` call?

File: src/components/MainPage/MainPage.tsx [hunk 2]
```diff
@@ -1,36 +1,29 @@
...
-                <span key={link.name} className={styles.header_link}>
-                    {link.name}
-                </span>
-            ))}
-        </div>
+        <NavigationMenu></NavigationMenu>
+
         <div className={styles.intro}>
-            <img className={styles.intro_image} src="https://via.placeholder.com/314" alt="Interesting" />
+            {/* <img className={styles.intro_image} src="https://via.placeholder.com/314" alt="Interesting" /> */}
```
Comment (WhiterDig) [inline src/components/MainPage/MainPage.tsx]: We can just remove this, I don't think we need the sample image anymore

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        handleCloseNavMenu();
+
+        // Navigate to selected page
+    };
+
+    return (
+        <AppBar className={styles.navigation_appbar}>
+            <Toolbar disableGutters className={styles.header_toolbar}>
+                {/* Collapsed Navigation Hamburger Menu for Smaller Screens */}
+                <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
```
Comment (WhiterDig) [inline src/components/NavigationMenu/NavigationMenu.tsx:29]: Love this! We could also use media query css to do this

Comment (WhiterDig) [reply src/components/NavigationMenu/NavigationMenu.tsx:29]: We don't have to change it, but I wanted to point that out

**Review by WhiterDig — COMMENTED** [review]
Love the work, it looks great. Nice thinking on the collapsed nav menu for smaller screens!

File: .eslintrc.js (hunk 1 above)
Comment (neonseal) [reply .eslintrc.js]: It was. Alternatively, we can leave the ignore to the `reportWebVitals()` call and return this rule to an error, which is probably safer.

File: src/components/NavigationMenu/NavigationMenu.tsx (hunk 3 above)
Comment (neonseal) [reply src/components/NavigationMenu/NavigationMenu.tsx:29]: Yeah, I built this largely basing off an example straight from the React Material UI documentation. So, this is how they seem to like to do media queries. I'm fine either way. I think media queries are easier to comprehend, but this seems more flexible.

**Review by WhiterDig — APPROVED** [review]

**Pull request merged**